### PR TITLE
test: Expand coverage of sourcemap validator

### PR DESCRIPTION
**Pull request by Gudahtt**
## **Description**

The sourcemap validator script has been updated to cover all build files. This better protects against sourcemap errors.

The file `offscreen-0.js` is optionally validated. It is expected to be missing in MV2 builds.

One file is still skipped because validation is failing (`sentry-install.js`).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25115?quickstart=1)

## **Related issues**

N/A

## **Manual testing steps**

* Run `yarn dist`
* Run `yarn validate-source-maps`, and see that it succeeds.

## **Screenshots/Recordings**

N/A

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: development/sourcemap-validator.js [hunk 1]
```diff
@@ -55,7 +68,10 @@ async function validateSourcemapForFile({ buildName }) {
   } catch (_) {
     // empty
   }
-  if (!rawBuild) {
+  if (!rawBuild)
```
Comment (Gudahtt) [inline development/sourcemap-validator.js:71]: Fix typo
```suggestion
  if (!rawBuild) {
```

**Comment by hjetpoluru** [comment]
@Gudahtt, I am getting below error when command was executed

**Comment by Gudahtt** [comment]
Sorry I forgot to clarify, that command should be run after a build (e.g. `yarn dist`). I will update the instructions in the PR description to say that.

**Comment by hjetpoluru** [comment]
ok thanks its working now

**Review by hjetpoluru — APPROVED** [review]
LGTM

File: development/sourcemap-validator.js [hunk 2]
```diff
@@ -22,25 +22,38 @@ start().catch((error) => {
...
-    `ui-0.js`,
-    `scripts/contentscript.js`,
-    // `scripts/inpage.js`, skipped because the validator can't sample the inlined `scripts/inpage.js` script
+    'background-0.js',
+    'common-0.js',
+    'content-script-0.js',
+    'ui-0.js',
+    'scripts/contentscript.js',
+    'scripts/disable-console.js',
+    'scripts/policy-load.js',
```
Comment (danjm) [inline development/sourcemap-validator.js:31]: Should we include the offscreen document file as well?

Comment (Gudahtt) [reply development/sourcemap-validator.js:31]: We do, it's in the optional target files array below. It is not included in MV2 builds, so we only validate it conditionally

**Review by legobeat — APPROVED** [review]

**Pull request merged**